docs/configuring-playbook-cactus-comments.md

```diff
@@ -0,0 +1,43 @@
+# Setting up Cactus Comments (optional)
+
+The playbook can install and configure [Cactus Comments](https://cactus.chat) for you.
+
+Cactus Comments is a **federated comment system** built on Matrix. The role allows you to self-host the system.
+It respects your privacy, and puts you in control.
+
+See the project's [documentation](https://cactus.chat/docs/getting-started/introduction/) to learn what it
+does and why it might be useful to you.
+
+
+## Configuration
+
+Add the following block to your `vars.yaml` and make sure to exchange the tokens to randomly generated values.
+
+```ỳaml
+#################
+## Cactus Chat ##
+#################
+
+matrix_cactus_comments_enabled: true
+matrix_cactus_comments_hs_token: wXchqGY94uhlk4OQ2ObPf0jsOMsnqXE9xuS2szB5
+matrix_cactus_comments_as_token: cvv8Y6OhAJqDbd1KCpRhc4mecZE9xuS2skBopaU5
+
+# To allow guest comments without users needing to log in, you need to have guest registration enabled.
+# To do this you need to uncomment one of the following lines (depending if you are using synapse or dentrite as a homeserver)
+# If you don't know which one you use: The default is synapse ;)
+# matrix_synapse_allow_guest_access: true
+# matrix_dentrite_allow_guest_access
+```
+
+## Installing
+
+After configuring the playbook, run the [installation](installing.md) command again:
+
+```
+ansible-playbook -i inventory/hosts setup.yml --tags=setup-all,start
+```
+
+
+## Usage
+
+To learn how to use cactus comments message @cactusbot:your-homeserver.com and type `help` or have a look in the [documentation](https://cactus.chat/docs/getting-started/quick-start/).
```

Comment:
Maybe this needs to be expanded.

When self-hosting is one still supposed to be embedding the files like this:

```html
<script type="text/javascript" src="https://latest.cactus.chat/cactus.js"></script>
<link rel="stylesheet" href="https://latest.cactus.chat/style.css" type="text/css">
```

or can one serve these files from the homeserver and be completely independent from `https://latest.cactus.chat`?

Besides the dependency, serving JS/CSS files from `latest.cactus.chat` while using an older version via the playbook will likely lead to various incompatibilities.

I believe we should serve everything from our own self-hosted instance of Cactus Comments.

Perhaps you need to inject some additional nginx configuration into `matrix-nginx-proxy` to proxy these files to the Cactus Comments container (which hopefully includes and can serve these files).

You can inject additional nginx configuration into any nginx vhost. Perhaps the `matrix.DOMAIN` vhost will be best, as it's always guaranteed to be available. See an example of how this can be done here: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/5f9f891322effa21d81c468cdcde701b6ba8c72c/roles/matrix-bridge-mautrix-telegram/tasks/init.yml#L30-L75

Reply:
Very good catch. My first idea (as the the container does not serve these files) is to download the files and let nginx serve them on the homeserver domain. An alternative would be to implement serving these files in upstream. The problem with that is, that the files are located in the client repo https://gitlab.com/cactus-comments/cactus-client/-/tree/main/src whereas we would serve it from https://gitlab.com/cactus-comments/cactus-appservice
I will not do this right away but think a bit more abut it...

Reply:
Just to clarify some stuff, so everyone's on the same page:
The Cactus appservice and client are basically completely decoupled - they only talk to Synapse, send spec'ed events and don't really know of each others existence. You can think of the client as a special-purpose Matrix client that renders a public Matrix room as a comment section, and the appservice is just there to make some stuff easy, like auto-creating rooms and replicating bans.

Downloading the files is fine to me. Alternatively, we can build a very boring nginx image in the client repo / ci pipeline upstream and use that. Whatever you prefer.

Reply:
Downloading the files is okay too. However, downloading from `https://latest.cactus.chat` means we've got a moving target and each time you run the playbook you're potentially getting new files.

I was wondering if we could download via `git` from some branch/tag.

Having a separate Docker image that serves the client files could also be nice. We can have pinned versions there, etc., and have the nginx configuration just proxy there. However, it's probably more difficult for you to create and publish such an image. Will it be multi-arch is another question.

Reply:
Ah, right. Should probably download from [the releases page](https://gitlab.com/cactus-comments/cactus-client/-/releases) then :)

Reply:
I now use the release or a local file if provided